contracts/PeeranhaCommunityToken.sol

```diff
@@ -0,0 +1,138 @@
+    communityTokenContainer.info.maxRewardPerPeriod = maxRewardPerPeriod;
+    communityTokenContainer.info.activeUsersInPeriod = activeUsersInPeriod;
+    communityTokenContainer.info.maxRewardPerUser = maxRewardPerPeriod / activeUsersInPeriod;
+  }
+
+  function getBalance() public view override returns(uint256) {   // public?
+    uint256 balance = IERC20Upgradeable(communityTokenContainer.info.tokenAddress).balanceOf(address(this));
+    require(balance >= communityTokenContainer.info.frezeTokens, "error_balance"); // todo: test??
+
+    return balance - communityTokenContainer.info.frezeTokens;
+  }
+
+  // get pool
+  function getTotalPeriodReward(uint16 period) private view returns(uint256) {
+    return communityTokenContainer.periodPool[period];
+  }
+
+  // set pool
+  function setTotalPeriodReward(RewardLib.PeriodRewardShares memory periodRewardShares, uint16 period) external override {
+    require(_msgSender() == communityTokenContainer.info.peeranhaCommunityTokenFactoryAddress, "only_community_token_factory_can_call_this_action");
+
+    uint256 totalPeriodReward = communityTokenContainer.info.maxRewardPerPeriod * 10 ** IERC20MetadataUpgradeable(communityTokenContainer.info.tokenAddress).decimals();
+    if (periodRewardShares.activeUsersInPeriod.length <= communityTokenContainer.info.activeUsersInPeriod) {
+      uint256 maxPeriodRewardPerUser = periodRewardShares.activeUsersInPeriod.length * communityTokenContainer.info.maxRewardPerUser * 10 ** IERC20MetadataUpgradeable(communityTokenContainer.info.tokenAddress).decimals();   // min?
+      totalPeriodReward = CommonLib.minUint256(totalPeriodReward, maxPeriodRewardPerUser);
+    }
+    totalPeriodReward = CommonLib.minUint256(totalPeriodReward, getBalance());
+    communityTokenContainer.info.frezeTokens += totalPeriodReward;  // todo: tests
+    
+    communityTokenContainer.periodPool[period] = totalPeriodReward;
+  }
+
+  function getUserCommunityReward(RewardLib.PeriodRewardShares memory periodRewardShares, uint32 ratingToReward, uint16 period) public view override returns(uint256) {
+    uint256 poolToken = getTotalPeriodReward(period);
+    uint256 userReward = getUserReward(periodRewardShares, ratingToReward * 1000, poolToken);
+
+    return userReward;
+  }
+
+  function payCommunityReward(RewardLib.PeriodRewardShares memory periodRewardShares, address userAddress, uint32 ratingToReward, uint16 period) external override {
+    uint256 userReward = getUserCommunityReward(periodRewardShares, ratingToReward, period);
+
+    if (userReward > 0) {
+      IERC20MetadataUpgradeable(communityTokenContainer.info.tokenAddress).transfer(userAddress, userReward);
+      communityTokenContainer.info.frezeTokens -= userReward;   // todo: tests
+    }
+  }
+
+  function getUserReward(RewardLib.PeriodRewardShares memory periodRewardShares, uint32 ratingToReward, uint256 poolToken) private pure returns(uint256) {
```

Comment:
`poolToken` -> `totalPeriodReward `
and not sure if separate function is needed because it is short and used only once.